src/commands/config/add_connection/mod.rs

```diff
@@ -272,4 +282,29 @@ impl AddNetworkConnection {
             Ok(None)
         }
     }
+
+    fn input_coingecko_url(
+        _context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<Option<crate::types::url::Url>> {
+        eprintln!();
+        #[derive(strum_macros::Display)]
+        enum ConfirmOptions {
+            #[strum(to_string = "Yes, I want to enter the coingecko API url")]
+            Yes,
+            #[strum(to_string = "No, I don't want to enter the coingecko API url")]
+            No,
+        }
+        let select_choose_input = Select::new(
+            "Do you want to enter the coingecko API url?",
+            vec![ConfirmOptions::Yes, ConfirmOptions::No],
+        )
+        .prompt()?;
+        if let ConfirmOptions::Yes = select_choose_input {
+            let coingecko_api: crate::types::url::Url =
+                CustomType::new("What is the coingecko API url?").prompt()?;
```

Comment:
Use [`.with_starting_input()`](https://docs.rs/inquire/latest/inquire/struct.CustomType.html#method.with_starting_input) to prepopulate the input with the default coingecko API URL.